### PR TITLE
docs: entity-manager written documentation PTL-1513

### PR DESCRIPTION
**Pull request by SzymonZur**
ATTENTION! NEW DOCS STRUCTURE
FOR THE NEXT THREE WEEKS, ONLY BRANCH FROM new-docs-structure
OTHERWISE, WAIT UNTIL THE NEW STRUCTURE HAS BEEN RELEASED

  - Please check the [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4) for details
  
__________

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?

Have you done a trial build to check all new or changed links?

Is there anything else you would like us to know?

**This week's exciting advice from the style guide**

- Write your headings in sentence case:

  - A good example
    This is a correct heading.
  - A Bad Example
    This is not a correct heading.

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+  - grid options
+  - user interface
+  - data display
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+
+The Grid is a key component of the **Entity Management**, enabling you to display and manage data from back-end resources. It offers a flexible, structured layout where each row represents an entity, and each column represents an attribute of that entity. The grid can be easily configured to pull data from specified back-end resources and is automatically populated with relevant information.
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx:28]:
```suggestion
The Entity Manager Grid is a key component of **Entity Management**, enabling you to display and manage data from back-end resources. It offers a flexible, structured layout where each row represents an entity, and each column represents an attribute of that entity. The grid can be easily configured to pull data from specified back-end resources and is automatically populated and updated by data from the backend resource.
```

Comment (patrickoneill-genesis) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx:28]: `and updated by data from the backend resource` maybe this bit could be improve too. Impress that any changes made to the backend are automatically streamed to the client and reflected in the view

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+If you want to control the appearance of the grid, use the `columns` attribute. This enables you to supply an array of `ColDef[]` properties to configure the columns that are displayed, their width and many other properties; these are the same properties that are used by the [Grid Pro](../003_grid-pro/index.mdx) component.
+
+The example array below creates six columns, specifying the width of each one. 
+- The `field` is the field in the data source.
+- The `headerName` is the text displayed in the column heading.
+- Three of the fields use `cellStyle` to control the alignment of the values displayed. 
+- One field (whose column heading is "Status") also uses `cellStyle` to control the colour of the values displayed. 
+
+```typescript
+const columnDefs: ColDef[] = [
```
Comment (patrickoneill-genesis) [inline docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx:44]: can we have a kotlin section describing the table used here?

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+
+The Grid is a key component of the **Entity Management**, enabling you to display and manage data from back-end resources. It offers a flexible, structured layout where each row represents an entity, and each column represents an attribute of that entity. The grid can be easily configured to pull data from specified back-end resources and is automatically populated with relevant information.
+
+This section explains how to configure and customize the grid to suit your application's needs.
+
+## Examples 
+
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx:33]: I think this title is unnecessary in the case where the entire file is just a set of examples, same for the other pages too

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx [hunk 4]
```diff
@@ -11,297 +11,340 @@ tags:
...
-:::info
-For more info on `persist-column-state-key` see [the section in the entity management.](#persist-column-state)
-:::
+### Common attributes & properties
 
-## Profile Management
+The most common attributes and properties you can configure for `entity-management` are listed below.
 
-The Profile Management micro front-end is used to manage the profiles on the front end. Two core components are used to manage the profiles - the grid and the form.
+| Attribute                  | Description                                          |   Mandatory |
```
[the remark this answers is not in the document]
Comment (matteematt) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx:311]: I see that later on you have them on a differnt page fully written out, perhaps just make sure you link to that from here and say "to see a full list of attribues with example see this section"

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx [hunk 5]
```diff
@@ -11,297 +11,340 @@ tags:
...
 
-- `INSERT_PROFILE` - the user can add new profiles
-- `DELETE_PROFILE` - the user can delete profiles
-- `AMEND_PROFILE` - the user can update existing profiles
+If you set multiple `entity-management` components in your application, **you must use unique keys** to persist the state - otherwise, the user experience will be unpredictable.
+:::
 
-#### Persist column state
-`persist-column-state-key` is a string value that is used to control how the column states are persisted through actions such as page changes or refreshes. If no `persist-column-state-key` property is set, then the behaviour will be to _not_ persist the column state, and the grid will revert back to its default state every time the user navigates away from it.
+| Property                  | Description                                          |   Mandatory |
```
[the remark this answers is not in the document]
Comment (matteematt) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx:333]: I see that later on you have them on a differnt page fully written out, perhaps just make sure you link to that from here and say "to see a full list of proprties with example see this section"

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/03_entity-manager-api.mdx
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/03_entity-manager-api.mdx:1]: This page is great, the gold standard of what we should be aiming for. Well done

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx (hunk 1 above)
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx:28]: I'll update the ending, but the beginning: "The Entity Manager Grid is a key component of **Entity Management**" - doesn't sound the best in my opinion

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/02_entity-manager-form.mdx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+- **`createFormUiSchema`** - Use this attribute when you need a custom form schema for creating a new record. It allows you to specify the fields, validation rules, and layout for the form when a user is adding a new entity. By providing a tailored UI schema, you can control the user experience and ensure that the form is optimized for new record creation.
+
+- **`updateFormUiSchema`** - Similarly, the `updateFormUiSchema` attribute is used when modifying an existing record. This allows you to define a schema specifically for editing an entity, which may differ from the creation form in terms of field visibility, default values, or validation rules. For example, you may want to disable certain fields or pre-populate them with existing data when updating an entity.
+
+If you're building a system where users can either add new entries or edit existing ones, using these attributes ensures that the forms are displayed and behave as required for each scenario. For instance, a "Create" form may have all fields enabled for user input, while an "Update" form may pre-fill certain fields and only allow modifications to specific attributes.
+
+```typescript
```
[the remark this answers is not in the document]
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/02_entity-manager-form.mdx:73]: Create schema is first at the top? where do you want to move it?

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx:333]: Info it's already there  

Screenshot 2024-11-26 at 21 03 38

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx (hunk 1 above)
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx]: Addressed in: [chore: addressed PR feedback PTL-1513](https://github.com/genesiscommunitysuccess/docs/pull/1951/commits/8e0a48c144037b54d8e8856d88f850cd48e063d9)

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx (hunk 2 above)
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx:44]: [chore: addressed PR feedback PTL-1513](https://github.com/genesiscommunitysuccess/docs/pull/1951/commits/8e0a48c144037b54d8e8856d88f850cd48e063d9) - I added simple server configuration but in index.md for COUNTERPARTY table

Here we only have column configuration, it is focused only on client-side.
Later, when we unify the examples throughout the docs, we can change and improve them, but for now it seems to be enough.

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx [hunk 7]
```diff
@@ -11,297 +11,340 @@ tags:
...
-![Example user management grid](./docs/img/foundation-user-management.png)
+Declaration:
+```html
+<entity-management></entity-management>
+```
 
-### Set-up
+Usage:
+```typescript
+const counterpartyUIschema = {
```
[the remark this answers is not in the document]
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx:164]: Done in [chore: addressed PR feedback PTL-1513](https://github.com/genesiscommunitysuccess/docs/pull/1951/commits/8e0a48c144037b54d8e8856d88f850cd48e063d9)

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx
[the remark this answers is not in the document]
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/index.mdx:1]: Done in [chore: addressed PR feedback PTL-1513](https://github.com/genesiscommunitysuccess/docs/pull/1951/commits/8e0a48c144037b54d8e8856d88f850cd48e063d9)

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx (hunk 3 above)
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/01_entity-manager-grid.mdx]: Done in [chore: addressed PR feedback PTL-1513](https://github.com/genesiscommunitysuccess/docs/pull/1951/commits/8e0a48c144037b54d8e8856d88f850cd48e063d9)

**Review by matteematt — APPROVED** [review]
Thanks for making the changes - it's an approval from me, though I see there is an outstanding item from Patrick. 

As discussed long term we want to use the same field defintions for all UI examples, for for now this is good enough for the 5th 🚀

File: docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/02_entity-manager-form.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (patrickoneill-genesis) [reply docs/001_develop/03_client-capabilities/005_grids/002_entity-manager/02_entity-manager-form.mdx:71]: Next to the section, it's only a nitpick.

**Review by patrickoneill-genesis — APPROVED** [review]

**Pull request merged**